Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +286,88 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
     SENTRY_LOG_DEBUG(@"SentryNetworkTracker finished HTTP span for sessionTask");
 }
 
+- (void)captureEvent:(NSURLSessionTask *)sessionTask
+{
+    NSInteger responseStatusCode = [self urlResponseStatusCode:sessionTask.response];
+
+    // TODO: check the string contains and regex
+    if (!self.isCaptureFailedRequests) {
+        return;
+    }
+
+    // TODO: check the range
+    if (responseStatusCode == 201) {
+        return;
+    }
+
+    NSString *message = [NSString
+        stringWithFormat:@"HTTP Client Error with status code: %li", (long)(responseStatusCode)];
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
+    NSArray<SentryThread *> *threads = [threadInspector getCurrentThreadsWithStackTrace];
```

Comment:
h: This function pauses the entire app to get every thread stackTrace, I think we should use `getCurrentThreads`, but then you only get the stacktrace from the current thread, which may not be the main thread.

But I would argue that we don't need stacktrace at all for a failed http request, IMO it won't add any value.

Reply:
Sounds good, I was not aware of `getCurrentThreads`, changing it, thanks.

Reply:
@brustolin `getCurrentThreads` does not return stack traces at all, I need at least from the current thread.

Reply:
The result of the request is always async, It runs in it own thread and most likely there is no stack, only one entry.

Reply:
`getCurrentThreads` returns multiple threads but no stack trace for the calling thread, I feel like having at least one stack trace is useful (current thread), none not that much, but the cost of reading all the stack traces from all threads is also too much, recommendation?

Reply:
Why do you think stacktrace is important for this issue? What it will help solve?

Reply:
Events are grouped per stack trace at least, otherwise, each event will be a new group, if that helps, likely not, but I don't have another option, I mentioned that in the Spec.

Reply:
> getCurrentThreads returns multiple threads but no stack trace for the calling thread

Are you sure about this? It should, otherwise, ANR would be broken.

> but the cost of reading all the stack traces from all threads is also too much, recommendation?

I think the cost is not that big, and this code is executed on a background thread cause it's networking, so I think it's fine.

Reply:
We already fixed this, the problem was that I was reading the first item of the `getCurrentThreads` list, as the ANR did, but the current thread isn't the pos=0.